apidef/api_definitions.go

```diff
@@ -782,8 +782,9 @@ type GraphQLSubgraphEntity struct {
 }
 
 type GraphQLEngineConfig struct {
-	FieldConfigs []GraphQLFieldConfig      `bson:"field_configs" json:"field_configs"`
-	DataSources  []GraphQLEngineDataSource `bson:"data_sources" json:"data_sources"`
+	FieldConfigs  []GraphQLFieldConfig      `bson:"field_configs" json:"field_configs"`
+	DataSources   []GraphQLEngineDataSource `bson:"data_sources" json:"data_sources"`
+	GlobalHeaders map[string]string         `bson:"global_headers" json:"global_headers"`
```

Comment:
we also need to extend `schema.go`:
https://github.com/TykTechnologies/tyk/blob/master/apidef/schema.go

Reply:
Also `map[string]string` may not be the perfect type if we want to make this extensible for the future.
Better would be a struct:
```
GlobalHeaders  []UDGGlobalHeader

type UDGGlobalHeader struct {
  Key string
  Value string
  // Additional data in future, e.g.
  // EnabledForAll bool
}

// As JSON
// { "global_headers": [ { "key": "Authorization", "value": "ey123" } ]
```

or also possible (maybe better for frontend):
```
GlobalHeaders  map[string]UDGGlobalHeaderValue

type UDGGlobalHeaderValue struct {
  Value string
  // Additional data in future, e.g.
  // EnabledForAll bool
}

// As JSON
// { "global_headers": { "Authorization": { "value": "ey123" } }
```